### PR TITLE
(pre-1138) Tidy up

### PR DESCRIPTION
**Pull request by yndajas**
## Changes in this PR

A few tidy ups made in the process of writing #409, including:
- updating Prison API types (this will hopefully stop being as much of a thing in random PRs after #348 is in
- mocking in unit tests
- a factory helper
- simplifying and tightening integration tests

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

**Comment by yndajas** [comment]
[Allow users to paginate through caselist (L)](https://trello.com/c/EQgr73Xp/1138-allow-users-to-paginate-through-caselist-l)

File: server/testutils/factories/factoryHelpers.ts [hunk 1]
```diff
@@ -8,6 +8,16 @@ export default class FactoryHelpers {
     return factory.buildList(quantity)
   }
 
+  static buildListWith<T>(factory: Factory<T>, properties: DeepPartial<T>, count: number): Array<T> {
```
Comment (andrewcharnley) [inline server/testutils/factories/factoryHelpers.ts:11]: Array
 .from({ length })
 .map ...

**Review by andrewcharnley — APPROVED** [review]

File: server/testutils/factories/factoryHelpers.ts (hunk 1 above)
Comment (yndajas) [reply server/testutils/factories/factoryHelpers.ts:11]: `Array.from({ length })` - looks like this creates an empty array?

Comment (yndajas) [reply server/testutils/factories/factoryHelpers.ts:11]: Could probably do `Array.from(Array(5)).map` but it's a bit hacky

Comment (andrewcharnley) [reply server/testutils/factories/factoryHelpers.ts:11]: No, just set { length:5 }, unlike new Array(5) it's enumerable. Your approach is fine though,

**Pull request merged**